src/dsp_tools/commands/xmllib/models/resource.py

```diff
@@ -0,0 +1,175 @@
+from __future__ import annotations
+
+import warnings
+from collections import defaultdict
+from dataclasses import dataclass
+from dataclasses import field
+
+from lxml import etree
+
+from dsp_tools.commands.xmllib.models.values import IntValue
+from dsp_tools.commands.xmllib.models.values import LinkValue
+from dsp_tools.commands.xmllib.models.values import SimpleText
+from dsp_tools.commands.xmllib.models.values import Value
+from dsp_tools.commands.xmllib.utils import is_string
+from dsp_tools.models.custom_warnings import DspToolsUserWarning
+from dsp_tools.models.exceptions import InputError
+from dsp_tools.utils.uri_util import is_iiif_uri
+
+XML_NAMESPACE_MAP = {None: "https://dasch.swiss/schema", "xsi": "http://www.w3.org/2001/XMLSchema-instance"}
+DASCH_SCHEMA = "{https://dasch.swiss/schema}"
+
+
+@dataclass
+class ResourceCollection:
+    resources: dict[str, Resource] = field(default_factory=dict)
```

Comment:
I would simply model this as a list of resources

Reply:
I modeled it as a dictionary because I want to be able to easily add further values to an existing resource at any time in the code. If it were a list I belive the code to add a value to an existing resource is more complicated. It is also easier to check if an id already exists in my opinion. What is exactly the reason why you prefer a list?

Reply:
to keep it simple and because I do not think it should be the concern of the `ResourceCollection` to modify resources. In fact, I very much believe that modifying existing data is an anti-pattern that should be avoided and discouraged as much as possible. In that sense, anything that should be added through the fluent interface should be "finished" and not need any modification.  
`ResourceCollection` as I envisioned it, should be an implementation detail of the fluent interface, that users ideally shouldn't ever notice it exists. Exposing any functionality beyond what the fluent interface needs, works against that, and then the users find themselves trying to understand what "the deal" is with this resource collection class, that in their eyes could just as well be a list/dict.